src/redux/books/Books.js

```diff
@@ -13,13 +16,34 @@ export const removeBook = (payload) => ({
   payload,
 });
 
+export const loadBook = (payload) => ({
+  type: GET_BOOK,
+  payload,
+});
+
+export const loadBooks = () => (dispatch) => {
+  getBooks().then((books) => {
+    Object.keys(books).forEach((ID) => {
+      const booksId = {};
+      booksId.id = ID;
+      booksId.title = books[ID][0].title;
+      booksId.category = books[ID][0].category;
+      dispatch(loadBook(booksId));
+    });
+  });
+};
+
 const reducer = (state = initialState, action) => {
   switch (action.type) {
     case ADD_BOOK:
+      newBook(action.payload);
       return [...state, action.payload];
 
     case REMOVE_BOOK:
+      deleteBook(action.payload);
       return state.filter((book) => book.id !== action.payload);
+    case GET_BOOK:
+      return [...state, action.payload];
```

Comment:
- [ ] I think one important rule about reducers and their cases it's that it must remain as a pure function, to achieve this we avoid generating side effects from it and we get the same result each time with the same input. Think "What happens if fetching the API is not available?"

That's why in this case it is recommended to use thunks to handle all async-related logic to our redux. Let's move these functions to a specific thunk and keep our reducer pure. [Read more about pure functions here.](https://www.sitepoint.com/functional-programming-pure-functions/#:~:text=A%20pure%20function%20is%20a,it%20does%20not%20change%20x%20.)